plugins/blip/modalday/SMBGBox.js

```diff
@@ -97,18 +238,34 @@ d3.chart('SMBGBoxOverlay', {
     var binKeys = Object.keys(binned);
     var retData = [];
     var value = function(d) { return d.value; };
+    var normalTime = function(d) { return new Date(d.normalTime); };
     var reduceForMean = function(s, n) { return s + n.value; };
     for (var i = 0; i < binKeys.length; ++i) {
+      var msX = parseInt(binKeys[i], 10);
       retData.push({
+        id: i,
```

Comment:
Setting the `id` to the index doesn't work in all circumstances. I was just looking at your updates running locally and when there aren't data points that actually fall into all 8 bins, setting _this_ `id` to the index clashes with expectations elsewhere in the code that all 8 bins always exist. Basically the bug is that you end up popping up an average tooltip for the bin to the left of the one you're hovering when one is missing... (I have a screenshot, but GitHub isn't letting me paste it on my poor WiFi, I'll try to send via e-mail instead.)